integreat_cms/static/src/js/search-query.ts

```diff
@@ -78,36 +88,94 @@ export const setSearchQueryEventListeners = () => {
         );
     });
 
-    const tableSearch = document.getElementById("table-search");
+    tableSearchInput.addEventListener("keydown", (event) => {
+        const suggestions = Array.from(suggestionList?.children || []) as HTMLElement[];
+        const input = event.target as HTMLInputElement;
+
+        if (suggestions.length === 0) {
+            return;
+        }
+
+        if (event.key === "ArrowLeft") {
+            event.preventDefault();
+            if (input.selectionStart !== null && input.selectionEnd !== null) {
+                const newCaretPosition = Math.max(0, input.selectionStart - 1);
+                input.setSelectionRange(newCaretPosition, newCaretPosition);
+            }
+        } else if (event.key === "ArrowRight") {
+            event.preventDefault();
+            if (input.selectionStart !== null && input.selectionEnd !== null) {
+                const newCaretPosition = Math.min(input.value.length, input.selectionEnd + 1);
+                input.setSelectionRange(newCaretPosition, newCaretPosition);
+            }
+        }
```

Comment:
OK, this definitely works. However, it looks like the reason this is required in the first place is this code snippet:

```js
    document.addEventListener("keydown", (event) => {
        if (event.key === "ArrowRight") {
            event.preventDefault();
            selectNextTimelineItem();
        } else if (event.key === "ArrowLeft") {
            event.preventDefault();
            selectPrevTimelineItem();
        }
    });
```
in [this file](https://github.com/digitalfabrik/integreat-cms/blob/develop/integreat_cms/static/src/js/revisions.ts#L308-L316).

I think that old code is a mistake. Looks like the intention was to prevent accidentally changing what translation version is being viewed in the version history timeline. But instead, it deactivates the arrow keys globally.

(The fact that we add ALL of our JS to EVERY page is a problematic story all on its own... 😂)

Rather than overwriting the behavior with your new code, I think it would be better to fix the old behavior. For example, you could add a check in `revisions.ts` to see if `.timeline-item` dics exist on the page, and only then add the arrow-key blocking code.

Reply:
@charludo thank you for pointing out this one. Because I was really wondering and was not able to find the reason behind it. It turns out that this exact code was blocking the default behaviour of other fields :crying_cat_face:

Reply:
@charludo Don't you think it would be better and more accessible if the timeline versions were also available with left and right arrows? Because i find it really easy just to choose the necessary version without clicking on every version... :thinking:

Reply:
I can only imagine it was added to prevent accidental selections, but yes, I am not sure how much sense that makes... 😅